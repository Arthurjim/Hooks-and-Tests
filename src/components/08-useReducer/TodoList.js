import React from "react";
import propTypes from "prop-types";
import TodoListItem from "./TodoListItem";

const TodoList = ({todos,deleteTodo,handleToogle}) => {
    return (
        <ul className="list-group list-group-flush">
            {todos.map((item, i) => (
                <TodoListItem key={item.id} item={item} i={i} handleToogle={handleToogle} deleteTodo={deleteTodo}/>
            ))}
        </ul>
    );
};

TodoList.propTypes = {
    todos: propTypes.array.isRequired,
    deleteTodo: propTypes.func.isRequired,
    handleToogle: propTypes.func.isRequired
}
TodoList.defaultProps = {
    todos:[]
}

export default TodoList;
