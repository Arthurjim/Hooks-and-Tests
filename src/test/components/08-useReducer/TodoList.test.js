const { shallow } = require("enzyme")
import TodoList from '../../../components/08-useReducer/TodoList'
import { demoTodos } from '../../fixtures/demoTodos'

describe('Pruebas en <TodoList />', () => {
    const deleteTodo = jest.fn()
   const handleToogle = jest.fn()
   const wrapper = shallow(<TodoList todos={demoTodos} deleteTodo={deleteTodo} handleToogle={handleToogle} />)
    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('Debe de tener todos los TodoListItem que se encuentren en el array ', () => {
        // expect(wrapper.find('TodoListItem'))
        expect(wrapper.find('TodoListItem').length).toBe(demoTodos.length);
        expect(wrapper.find('TodoListItem').at(0).prop('deleteTodo')).toEqual(expect.any(Function))
    })
    
    
})
