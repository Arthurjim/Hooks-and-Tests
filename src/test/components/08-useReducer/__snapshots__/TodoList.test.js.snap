// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <TodoList /> Debe de mostrarse correctamente 1`] = `
<ul
  className="list-group list-group-flush"
>
  <TodoListItem
    deleteTodo={[MockFunction]}
    handleToogle={[MockFunction]}
    i={0}
    item={
      Object {
        "desc": "Tareas",
        "done": false,
        "id": 1,
      }
    }
    key="1"
  />
  <TodoListItem
    deleteTodo={[MockFunction]}
    handleToogle={[MockFunction]}
    i={1}
    item={
      Object {
        "desc": "Aprender node",
        "done": false,
        "id": 2,
      }
    }
    key="2"
  />
</ul>
`;
